src/atomWithQuery.ts

```diff
@@ -0,0 +1,30 @@
+import { QueryClient, QueryObserver } from '@tanstack/query-core'
+import type { QueryKey, QueryObserverOptions } from '@tanstack/query-core'
+import { type Getter, atom } from 'jotai/vanilla'
+import { queryClientAtom } from './queryClientAtom'
+
+export function atomWithQuery<
+  TQueryFnData = unknown,
+  TError = unknown,
+  TData = TQueryFnData,
+  TQueryData = TQueryFnData,
+  TQueryKey extends QueryKey = QueryKey,
+>(
+  getOptions: (
+    get: Getter
+  ) => QueryObserverOptions<TQueryFnData, TError, TData, TQueryData, TQueryKey>,
+  getQueryClient: (get: Getter) => QueryClient = (get) => get(queryClientAtom)
+) {
```

Comment:
What would be the return type? Can you explicitly type it? (For now, I don't care the impl below.)